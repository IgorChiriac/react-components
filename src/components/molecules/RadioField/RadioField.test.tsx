import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import InputCheckbox from './RadioField';

describe('<RadioField />', () => {
  it('renders the component with props with no a11y violations', async () => {
    const { container } = render(<InputCheckbox label={'hello'} inputProps={{ name: 'test1', value: 1 }} />);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
