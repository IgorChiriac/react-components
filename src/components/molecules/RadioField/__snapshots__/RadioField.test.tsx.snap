// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioField /> renders the component with props with no a11y violations 1`] = `
.c5 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0D0A38;
  line-height: 1.6;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-weight: 600;
  font-size: 16px;
}

.c4 {
  display: block;
}

.c1 {
  max-width: 100%;
  width: 100%;
}

.c0 {
  position: relative;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1.4;
  color: #0D0A38;
  font-weight: 400;
  position: relative;
}

.c3:before {
  content: '';
  background-color: #EAEAEE;
  border-radius: 50%;
  height: 24px;
  width: 24px;
  min-width: 24px;
  display: inline-block;
  vertical-align: top;
  margin-right: 8px;
  border: 2px #DCDBE2 solid;
}

.c3:hover {
  cursor: pointer;
}

.c3:hover:before {
  border: 2px solid #DCDBE2;
}

.c2 {
  width: 1px;
  height: 1px;
  opacity: 0;
  z-index: -1;
  position: absolute;
}

.c2:hover + label:before,
.c2:focus + label:before {
  border: 2px #4A3EDE solid;
}

.c2:checked + label {
  font-weight: 600;
}

.c2:checked + label:before {
  content: '';
  background-color: #4A3EDE;
  border: 2px #4A3EDE solid;
  border-radius: 50%;
  height: 24px;
  width: 24px;
  display: inline-block;
  vertical-align: top;
  margin-right: 8px;
}

.c2:checked + label:after {
  content: '';
  background-color: #FFFFFF;
  border-radius: 50%;
  height: 8px;
  width: 8px;
  display: inline-block;
  -webkit-animation: ciAMHu 120ms ease-in-out;
  animation: ciAMHu 120ms ease-in-out;
  left: 8px;
  top: 8px;
  position: absolute;
}

.c2:disabled + label {
  color: #DCDBE2;
}

.c2:disabled + label:before {
  border-color: #DCDBE2;
  background-color: #EAEAEE;
}

.c2:disabled + label:hover {
  cursor: not-allowed;
}

<div
  class="c0 c1"
>
  <input
    class="c2"
    id="radio-id-1"
    name="test1"
    type="radio"
    value="1"
  />
  <label
    class="c3 c4 c5"
    color="#0D0A38"
    for="radio-id-1"
  >
    hello
  </label>
</div>
`;
